src/main/java/data/implementations/EdgeWithCoordinates.java

```diff
@@ -0,0 +1,38 @@
+package data.implementations;
+
+public class EdgeWithCoordinates {
+
+    private int fromX;
+    private int fromY;
+    private int toX;
+    private int toY;
+    private int length;
+
+    public EdgeWithCoordinates(int fromX, int fromY, int toX, int toY, int length) {
```

Comment:
Czy length jest inną długością niż długość "kreski" pomiędzy podanymi punktami? Jeżeli nie, to przecież to się może "samo" wyliczyć. Nie trzeba tego podawać w konstruktorze.
Tak w ogóle, nie myślałeś o tym, by ta klasa np. rozszerzała Edge? Ewentualnie do konstruktora można by przekazywać dwa MapPointy i wewnątrz konstruktora wyciągać sobie x i y. Wtedy miałbyś tylko dwa argumenty zamiast 5. Przemyśl to, czy to pasuje do reszty.

Reply:
W sumie ta klasa została stworzona tylko na potrzeby szybszego rysowania edgów, więc nie ma potrzeby rozszerzania edga.
Rzeczywiście, to rozwiązanie z Pointami może być bardziej przejrzyste

Reply:
Albo jednak może lepiej zostawię tak jak jest, bo metoda "scaleCoords" w paintPanel zwraca tablicę przeskalowanych intów, więc jakbym zmieniał tego EdgeWithCoordinates, to też bym musiał zmienić tą metodę, a tu już Pointa przekazywać spoko, ale zwracać Pointa i promień (przeskalowaną wielkość noda, edga lub znacznika lokacji), już trudno. Więc może i byłoby przejrzyste, ale potem robić jakieś dziwne rzeczy z tymi pointami, to szkoda zachodu.

Reply:
Ten length powinien być jednak w konstruktorze, bo pionowe edge mają długość 1, a po obliczeniu wychodzą czasami jakieś niespodziewane wartość, no bo powinno być 0. Wychodzą jednak zupełnie inne np. 10, 3 czasami jeszcze inne, wynika to z tego, że mapy są przesunięte troszkę względem siebie. Więc zostawię ten length w konstruktorze.